cpp/src/groupby/sort/group_nunique.cu

```diff
@@ -28,7 +28,74 @@ namespace experimental {
+               rmm::device_vector<size_type> const &group_labels,
+               size_type num_groups,
+               rmm::device_vector<size_type> const &group_offsets,
+               include_nulls _include_nulls,
+               rmm::mr::device_memory_resource *mr, cudaStream_t stream) 
+    {
+      auto result = make_numeric_column(data_type(type_to_id<size_type>()), 
+                      num_groups,
+                      mask_state::UNALLOCATED, stream, mr);
+
+      if (num_groups == 0) {
+        return result;
+      }
+
+      auto values_view = column_device_view::create(values);
+      if (values.has_nulls()) {
+        auto comp = element_equality_comparator<true>{*values_view, *values_view};
+        auto is_unique_iterator = thrust::make_transform_iterator(
+            thrust::make_counting_iterator<size_type>(0),
+            [v = *values_view, comp, _include_nulls,
+             group_offsets = group_offsets.data().get(),
+             group_labels = group_labels.data().get()] 
+             __device__(auto i) -> size_type {
+              bool is_unique = (_include_nulls == include_nulls::YES ||
+                                v.is_valid_nocheck(i)) &&
+                               (group_offsets[group_labels[i]] == i ||
+                               (not comp.operator()<T>(i, i-1)));
+              return static_cast<size_type>(is_unique);
+            });
+
+        thrust::reduce_by_key(rmm::exec_policy(stream)->on(stream),
+                              group_labels.begin(), group_labels.end(),
+                              is_unique_iterator,
+                              thrust::make_discard_iterator(),
+                              result->mutable_view().begin<size_type>());
+      } else {
+        auto comp = element_equality_comparator<false>{*values_view, *values_view};
+        auto is_unique_iterator = thrust::make_transform_iterator(
+            thrust::make_counting_iterator<size_type>(0),
+            [v = *values_view, comp, group_offsets = group_offsets.data().get(),
+             group_labels = group_labels.data().get()] 
+             __device__(auto i) -> size_type {
+              bool is_unique = group_offsets[group_labels[i]] == i ||
+                               (not comp.operator()<T>(i, i - 1));
+              return static_cast<size_type>(is_unique);
+            });
+        thrust::reduce_by_key(rmm::exec_policy(stream)->on(stream),
+                              group_labels.begin(), group_labels.end(),
+                              is_unique_iterator,
+                              thrust::make_discard_iterator(),
+                              result->mutable_view().begin<size_type>());
+      }
+      return result;
+    }
+
+    template <typename T, typename... Args>
+    std::enable_if_t<!is_fixed_width<T>(), std::unique_ptr<column>>
+    operator()(Args &&... args) {
+      CUDF_FAIL("Only fixed width types are supported in nunique");
```

Comment:
Why? The comparator works for all types.

Reply:
Yes. it should work.